### PR TITLE
feat(cli): `wing gen-docs`

### LINKED ISSUE #6905: Generate API documentation for Wing libraries
### Use Case

I am building a Wing library and I'd like to generate documentation for all the public APIs.

### Proposed Solution

I think that the best approach would be to emit a `.jsii` manifest when compiling a library and then leverage our fork `jsii-docgen` to generate API docs.

(This will also have the incredible benefit of allowing us to use all the JSII toolchain to generate language bindings to all JSII supported languages - https://github.com/winglang/wing/issues/6904).

### Implementation Notes

_No response_

### Component

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.
* If this issue is labeled **needs-discussion**, it means the spec has not been finalized yet. Please reach out on the #dev channel in the [Wing Discord](https://t.winglang.io/discord).

### PR DESCRIPTION
**Pull request by Chriscbr**
Adds a new `gen-docs` subcommand that generates API documentation for your Wing library. Running `wing gen-docs` will generate an API.md file in your current directory.

Closes #6905

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: apps/wing/fixtures/valid1/API.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<h2>API Reference</h2>
+
+<h3>Table of Contents</h3>
+
+- **Classes**
+  - <a href="#valid1.Foo">Foo</a>
+
+<h3 id="valid1.Foo">Foo (preflight class)</h3>
```
Comment (Chriscbr) [inline apps/wing/fixtures/valid1/API.md:8]: Using these HTML header tags doesn't feel totally right, but it was the only way I could figure out to get the anchors working as expected.

File: apps/wing/src/cli.ts [hunk 2]
```diff
@@ -164,6 +164,12 @@ async function main() {
     .hook("preAction", collectAnalyticsHook)
     .action(runSubCommand("lsp"));
 
+  program
+    .command("generate-docs")
```
[the remark this answers is not in the document]
Comment (Chriscbr) [reply apps/wing/src/cli.ts]: We already have a `wing docs` command which opens your browser to winglang.io/docs 🤷

[the remark this answers is not in the document]
Comment (eladb) [reply apps/wing/src/cli.ts]: Oh, I see...

Is it useful? Maybe we can deprecate it and reuse it for doc generation, no?

[the remark this answers is not in the document]
Comment (Chriscbr) [reply apps/wing/src/cli.ts]: done!

**Pull request merged**